packages/components/src/dropdown/index.js

```diff
@@ -50,7 +50,7 @@ export default function Dropdown( props ) {
 				onToggle( false );
 			}
 		},
-		[]
+		[ onToggle ]
```

Comment:
This effect seems weird to me, in that it will call `onToggle( false )` on cleanup even when no toggle has happened 🤔 So I checked the history, and the [pre-refactor logic](https://github.com/WordPress/gutenberg/pull/23142/files#diff-dacdfb5a25d27771d335dc7307b037ce12d6b122b40b24d1800bd9bd0cd045f7L34-L36) includes an `isOpen` condition, which makes sense. The "cleanup" is that _if the dropdown is open at unmount time_, it should be toggled closed. I have a feeling this is a refactoring error.

But since we haven't gotten any complaints in two years, I have to assume nobody is using the `onToggle` for non-idempotent things 😅

Reply:
You make a good point. I can plan a followup to re-add that check! Thank you both!

Reply:
followup here: #41604